File: intro-to-cs50-harvardx/week3/collatz-conjecture.js
let steps = 0
function collatz (n) {
  if ( !Number.isInteger(n * 1) || n <= 0) throw new Error('n should be a positive integer')
  n = n * 1
  if (n === 1) return console.log('done with', steps)
  steps++
  n % 2 === 0 ? collatz(n / 2) : collatz(n * 3 + 1)
}

let arg = process.argv[2]
collatz(arg)

File: intro-to-cs50-harvardx/week3/collatz-conjecture.c
#include <stdio.h>

int collatz (int n, int steps);
int main () {
  int steps = 0;
  collatz(27, steps);
  return 0;
}

int collatz (int n, int steps) {
  if (n == 1) return printf("Done with %d steps\n", steps);
  steps++;
  return n % 2 == 0 ? collatz(n / 2, steps) : collatz(n * 3 + 1, steps);
}